src/TankController.cpp

```diff
@@ -158,25 +161,31 @@ void TankController::handleUI() {
 /**
  * This is one of two public instance functions.
  * It is called repeatedly while the board is on.
- * (It appears to be called about once every 15 ms.)
  */
 void TankController::loop(bool report_loop_delay) {
   static unsigned long lastTime = 0;
   unsigned long thisTime = millis();
   if (report_loop_delay && lastTime && thisTime - lastTime > 500) {
-    // report unusual delay
     serial(F("unexpected delay of %i ms"), thisTime - lastTime);
   }
-  lastTime = thisTime;
+  unsigned long start = millis();
   wdt_reset();
-  blink();                                // blink the on-board LED to show that we are running
-  updateControls();                       // turn CO2 and temperature controls on or off
-  handleUI();                             // look at keypad, update LCD (~90ms)
-  DataLogger::instance()->loop();         // record current data to SD and serial
-  GetTime::instance()->loop();            // update the time
-  PushingBox::instance()->loop();         // write data to Google Sheets (~1130ms every report)
-  Ethernet_TC::instance()->loop();        // renew DHCP lease
-  EthernetServer_TC::instance()->loop();  // handle any HTTP requests
+  blink();                                // blink the on-board LED to show that we are running (0ms)
+  updateControls();                       // turn CO2 and temperature controls on or off (~90ms)
+  handleUI();                             // look at keypad, update LCD (~10ms)
+  DataLogger::instance()->loop();         // record current data to SD and serial (~80ms)
+  GetTime::instance()->loop();            // update the time (~0ms)
+  PushingBox::instance()->loop();         // write data to Google Sheets (~0ms; ~1130ms every report)
+  Ethernet_TC::instance()->loop();        // renew DHCP lease (~0ms)
+  EthernetServer_TC::instance()->loop();  // handle any HTTP requests (~0ms)
+  if (report_loop_delay) {
+    static long int count = 0;
+    unsigned long loopTime = millis() - start;
+    if (++count % 10000 == 1 || loopTime > 200) {  // first time through and periodically thereafter
+      serial(F("TankController::loop() - took %lu ms (at %lu sec uptime)"), loopTime, start / 1000);
+    }
+    lastTime = millis();
+  }
```

Comment:
Here, just set `previousLoopStop` to `millis()` ("previous" isn't quite right, but it's about to be previous)